### PR TITLE
[esp/sim] move Simulator to sim namespace

### PR DESCRIPTION
**Pull request by msbaines**
Goal is to collapse Simulator and SimulatorWithAgents into one class.
Moved Simulator to sim namespace an in intermediate step. Next step
is to pull up SimulatorWithAgents logic into Simulator.

The change is part of the sensor refactoring.

## Motivation and Context

Collapse Simulator and SimulatorWithAgents into one class.

## How Has This Been Tested

SimTest

## Types of changes

- [X] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/esp/bindings/bindings.cpp [hunk 1]
```diff
@@ -69,6 +69,7 @@ PYBIND11_MODULE(habitat_sim_bindings, m) {
   esp::physics::initPhysicsBindings(m);
   esp::scene::initSceneBindings(m);
   esp::sensor::initSensorBindings(m);
+  esp::sim::initSimBindings(m);
```
Comment (erikwijmans) [inline src/esp/bindings/bindings.cpp]: I believe this needs to be moved before init sensor bindings for doc generation to cross-link correctly.

Comment (erikwijmans) [reply src/esp/bindings/bindings.cpp]: Actually, it looks like we have a lot of things that aren't cross linking correctly anymore :/

Comment (msbaines) [reply src/esp/bindings/bindings.cpp]: How does the cross-linking work? How do I verify it?

Comment (msbaines) [reply src/esp/bindings/bindings.cpp]: Talked to @erikwijmans offline. Will fix cross-linking in a separate change.

**Pull request merged**